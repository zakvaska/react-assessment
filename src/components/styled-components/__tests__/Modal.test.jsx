import React from 'react';
import { render } from '@testing-library/react';
import { 
  ModalWrapper,
  ModalDialog,
  ModalBody,
  ModalHeader,
  ModalFooter 
 } from '@components/styled-components/Task';
import 'jest-styled-components';

it('ModalHeader renders correctly', () => {
  const { container } = render(<ModalHeader>Hello</ModalHeader>);
  expect(container).toMatchSnapshot();
});

it('ModalBody renders correctly', () => {
  const { container } = render(<ModalBody>World</ModalBody>);
  expect(container).toMatchSnapshot();
});

it('ModalFooter renders correctly', () => {
  const { container } = render(<ModalFooter>Today</ModalFooter>);
  expect(container).toMatchSnapshot();
});

it('Modal renders correctly with callback', () => {
  const handleClose = jest.fn();
  const { container } = render(
    <ModalWrapper onClick={handleClose}>
      <ModalDialog>
        <ModalHeader>Hello</ModalHeader>
        <ModalBody>World</ModalBody>
        <ModalFooter>Today</ModalFooter>
      </ModalDialog>
    </ModalWrapper>
  );
  expect(container).toMatchSnapshot();
});

    