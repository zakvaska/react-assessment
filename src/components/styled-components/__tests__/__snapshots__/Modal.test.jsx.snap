// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly with callback 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c1 {
  background-color: white;
  border: 1px solid #d10720;
  width: 35rem;
  border-radius: .75rem;
  margin: 10rem auto;
  padding: 1.5rem;
  color: black;
  text-align: start;
  font-family: sans-serif;
  font-size: 1.1rem;
  cursor: default;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #e1e1e1;
}

.c2 > div {
  padding-left: .5rem;
  font-weight: bold;
  color: black;
  font-family: sans-serif;
}

.c3 {
  font-family: sans-serif;
  font-size: 1rem;
  padding-top: 1rem;
}

.c4 {
  padding-top: 1rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Hello
      </div>
      <div
        class="c3"
      >
        World
      </div>
      <div
        class="c4"
      >
        Today
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalBody renders correctly 1`] = `
.c0 {
  font-family: sans-serif;
  font-size: 1rem;
  padding-top: 1rem;
}

<div>
  <div
    class="c0"
  >
    World
  </div>
</div>
`;

exports[`ModalFooter renders correctly 1`] = `
.c0 {
  padding-top: 1rem;
}

<div>
  <div
    class="c0"
  >
    Today
  </div>
</div>
`;

exports[`ModalHeader renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #e1e1e1;
}

.c0 > div {
  padding-left: .5rem;
  font-weight: bold;
  color: black;
  font-family: sans-serif;
}

<div>
  <div
    class="c0"
  >
    Hello
  </div>
</div>
`;
